### PR TITLE
Menu animation

### PR DESCRIPTION
**Pull request by JannisPetschenka**
Small changes to the menubar to add some animation :)

### CODE REVIEW

**Comment by ErikReider** [comment]
I'll look at this tomorrow. Ping me if I forget :)

File: src/configSchema.json [hunk 1]
```diff
@@ -423,6 +423,17 @@
               "default": "right",
               "enum": ["right", "left"]
             },
+            "animation_type": {
+              "type": "string",
+							"default": "slide_down",
+							"description": "Animation type for menu",
+							"enum": ["slide_down", "slide_up", "none"]
```
Comment (ErikReider) [inline src/configSchema.json:430]: Wrong indent type

File: src/controlCenter/widgets/menubar/menubar.vala [hunk 2]
```diff
@@ -160,14 +167,42 @@ namespace SwayNotificationCenter.Widgets {
...
+                string ? animation_type = get_prop<string> (obj, "animation_type");
+                if (animation_type == null) {
+                    animation_type = "slide_down";
+                    info ("No animation-type for menu-object given using default");
+                }
+
+                Gtk.RevealerTransitionType revealer_type;
+
+                switch (animation_type) {
+                    case "none":
```
Comment (ErikReider) [inline src/controlCenter/widgets/menubar/menubar.vala:182]: Put default here to simplify

File: src/controlCenter/widgets/menubar/menubar.vala [hunk 3]
```diff
@@ -92,14 +94,19 @@ namespace SwayNotificationCenter.Widgets {
 
                     Gtk.Box menu = new Gtk.Box (Gtk.Orientation.VERTICAL, 0);
                     menu.get_style_context ().add_class (obj.name);
```
Comment (ErikReider) [inline src/controlCenter/widgets/menubar/menubar.vala:96]: Forgot to mention this before, we should avoid setting the class of the menu and the buttons if the `obj.name` is null. My bad 😅

File: src/controlCenter/widgets/menubar/menubar.vala [hunk 4]
```diff
@@ -92,14 +94,19 @@ namespace SwayNotificationCenter.Widgets {
...
                     show_button.clicked.connect (() => {
-                    bool visible = !menu.visible;
+                    bool visible = !r.get_reveal_child ();
                     foreach (var o in menu_objects) {
-                        o.menu ?.set_visible (false);
+                        o.revealer ?.set_reveal_child (false);
                     }
-                    menu.set_visible (visible);
+                    r.set_reveal_child (visible);
                 });
```
Comment (ErikReider) [inline src/controlCenter/widgets/menubar/menubar.vala:110]: Indent this. This is a really annoying Uncrustify issue that I've not been able to solve yet...

File: src/controlCenter/widgets/menubar/menubar.vala [hunk 5]
```diff
@@ -17,7 +17,9 @@ namespace SwayNotificationCenter.Widgets {
         string ? label;
         Position ? position;
         Action[] actions;
-        Gtk.Box ? menu;
+        Gtk.Revealer ? revealer;
+        int animation_duration;
+        Gtk.RevealerTransitionType ? animation_type;
```
Comment (ErikReider) [inline src/controlCenter/widgets/menubar/menubar.vala:22]: Remove the question mark. Shouldn't be nullable

**Review by ErikReider — CHANGES_REQUESTED** [review]
LGTM otherwise! :)

**Comment by ErikReider** [comment]
Oh yeah, also forgot to mention that there's a weird graphical issue when opening two menus. Looks like the 2nd menu opens over the first one. I'm not sure if that's due to them expanding/contracting at the same time resulting in that behavior or not. Just letting you know :)

**Comment by JannisPetschenka** [comment]
> Oh yeah, also forgot to mention that there's a weird graphical issue when opening two menus. Looks like the 2nd menu opens over the first one. I'm not sure if that's due to them expanding/contracting at the same time resulting in that behavior or not. Just letting you know :)

Yeah, I'm aware of that issue :) The only way to fix this I can think of is to somehow wait for `animation_duration` ms after closing the boxes before opening the correct one. But that would double the time it takes to open a drop-down :upside_down_face:.. So I thought about just calling it a feature rather than a bug :laughing: But I can look into changing it if you prefer :)

**Review by ErikReider — APPROVED** [review]

**Comment by ErikReider** [comment]
Thanks again for your help! :)

**Pull request merged**